go/test/endtoend/vtorc/api/api_test.go

```diff
@@ -156,23 +168,69 @@ func TestProblemsAPI(t *testing.T) {
 		assert.Contains(t, resp, fmt.Sprintf(`"InstanceAlias": "%v"`, replica.Alias))
 
 		// Check that filtering using keyspace and shard works
-		status, resp = utils.MakeAPICall(t, vtorc, "/api/problems?keyspace=ks&shard=0")
+		status, resp, err = utils.MakeAPICall(t, vtorc, "/api/problems?keyspace=ks&shard=0")
+		require.NoError(t, err)
 		assert.Equal(t, 200, status, resp)
 		assert.Contains(t, resp, fmt.Sprintf(`"InstanceAlias": "%v"`, replica.Alias))
 
 		// Check that filtering using keyspace works
-		status, resp = utils.MakeAPICall(t, vtorc, "/api/problems?keyspace=ks")
+		status, resp, err = utils.MakeAPICall(t, vtorc, "/api/problems?keyspace=ks")
+		require.NoError(t, err)
 		assert.Equal(t, 200, status, resp)
 		assert.Contains(t, resp, fmt.Sprintf(`"InstanceAlias": "%v"`, replica.Alias))
 
 		// Check that filtering using keyspace and shard works
-		status, resp = utils.MakeAPICall(t, vtorc, "/api/problems?keyspace=ks&shard=80-")
+		status, resp, err = utils.MakeAPICall(t, vtorc, "/api/problems?keyspace=ks&shard=80-")
+		require.NoError(t, err)
 		assert.Equal(t, 200, status, resp)
 		assert.Equal(t, "null", resp)
 
 		// Check that filtering using just the shard fails
-		status, resp = utils.MakeAPICall(t, vtorc, "/api/problems?shard=0")
+		status, resp, err = utils.MakeAPICall(t, vtorc, "/api/problems?shard=0")
+		require.NoError(t, err)
 		assert.Equal(t, 400, status, resp)
 		assert.Equal(t, "Filtering by shard without keyspace isn't supported\n", resp)
 	})
 }
+
+// TestIfDiscoveringHappenedLaterInHealthCheck checks that `IsDiscovering` flag in `HealthStatus` turns `true`, sometime
+// after flag `Healthy` turns `true`.
+func TestIfDiscoveringHappenedLaterInHealthCheck(t *testing.T) {
+	defer cluster.PanicHandler(t)
+	utils.SetupVttabletsAndVTOrcs(t, clusterInfo, 2, 1, nil, cluster.VTOrcConfiguration{
```

Comment:
FYI, I have found out VTOrcConfiguration is not getting translated to vtorc config. But this is a issue I would like to resolve in some other PR.